kvbc/include/db_adapter_interface.h

```diff
@@ -74,6 +75,7 @@ class IDbAdapter {
   // TODO [TK] not sure it's needed for long term
   virtual std::shared_ptr<storage::IDBClient> getDb() const = 0;
 
+  virtual void setAggregator(std::shared_ptr<concordMetrics::Aggregator> aggregator) = 0;
```

Comment:
I am thinking if we actually need this to be part of the base class. We can start by only adding it in the merkle DBAdapter and add to the base class in the future, if needed.

It might be done by passing the agregator to the `v2MerkleTree::RocksDBStorageFactory`'s constructor. The factory will then call a non-virtual `setAggregator()` method on the instances it produces.

Just something to consider to avoid further inheritance and adding methods to a base class.

Reply:
I think in the future we might want to add metrics also in the direct key value store (for example, to compare performance). So I think that would be good thing to have it with a default imp in the interface, what do you think?